src/wasm/wasm-type.cpp

```diff
@@ -521,8 +521,19 @@ bool Type::isSubType(Type left, Type right) {
     return true;
   }
   if (left.isRef() && right.isRef()) {
-    return right == Type::anyref ||
-           (left == Type::i31ref && right == Type::eqref);
+    if (right == Type::anyref) {
+      // Everything is a subtype of anyref.
+      return true;
+    }
+    if (left == Type::i31ref && right == Type::eqref) {
+      // i31 is a subtype of eqref.
+      return true;
+    }
+    if (right == Type::funcref && left.getHeapType().isSignature()) {
```

Comment:
```suggestion
    if (left.getHeapType().isSignature() && right == Type::funcref) {
```
I would put the left on the left and the right on the right.

Reply:
I guess there's no reason not to, yeah, there's no speed reason here (which is I think why my brain preferred it).